onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1545,6 +1619,9 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
       const std::string cache_path = GetCachePath(trt_state->engine_cache_path, trt_state->trt_node_name_with_precision);
       const std::string engine_cache_path = cache_path + ".engine";
       const std::string profile_cache_path = cache_path + ".profile";
+      cudaDeviceProp prop;
+      CUDA_CALL_THROW(cudaGetDeviceProperties(&prop, 0));
```

Comment:
I think we should check timeing cache enable flag first and then call CUDA API since this code is in inference/compute function where we want to reduce as many overhead as possible.